test/Microsoft.AspNet.Server.KestrelTests/ConnectionFilterTests.cs

```diff
@@ -80,7 +81,36 @@ public class ConnectionFilterTests
                         "Hello World!");
                 }
             }
-       }
+        }
+
+        [ConditionalFact]
+        [FrameworkSkipCondition(RuntimeFrameworks.Mono, SkipReason = "Test hangs after execution on Mono.")]
```

Comment:
Yeah, it looks like the problem is gone. I'd like to do a little more testing though.